src/sagas/pushNotification.js

```diff
@@ -0,0 +1,440 @@
+ *   txId: 'txId1',
+ * }
+ * @inner
+ */
+const messageHandler = async (message) => {
+  const { data } = message;
+  if (!localization.hasKey(data.titleLocKey)) {
+    console.debug('unknown message titleLocKey', data.titleLocKey);
+    return;
+  }
+
+  if (!localization.hasKey(data.bodyLocKey)) {
+    console.debug('unknown message bodyLocKey', data.bodyLocKey);
+    return;
+  }
+
+  const bodyArgs = JSON.parse(data.bodyLocArgs);
+  const title = localization.getMessage(data.titleLocKey);
+  const body = localization.getMessage(data.bodyLocKey, bodyArgs);
+
+  notifee.displayNotification({
+    title,
+    body,
+    android: {
+      channelId: TRANSACTION_CHANNEL_ID,
+    },
+  });
+};
+
+const confirmDeviceRegistrationOnFirebase = async () => {
+  try {
+    // Make sure deviceId is registered on the FCM
+    await messaging().registerDeviceForRemoteMessages();
+  } catch (error) {
+    console.error(`Error confirming the device is registered on FCM: ${error.message}`, error);
+  }
+};
+
+const installForegroundListener = () => {
+  try {
+    // Add listeners for push notifications on foreground and background
+    messaging().onMessage(onForegroundMessage);
+  } catch (error) {
+    console.error(`Error installing foreground listener to push notification: ${error.message}`, error);
+  }
+};
+
+const createChannelIfNotExists = async () => {
+  try {
+    const hasTransactionChannel = await notifee.isChannelCreated(TRANSACTION_CHANNEL_ID);
+    if (!hasTransactionChannel) {
+      await notifee.createChannel({
+        id: TRANSACTION_CHANNEL_ID,
+        name: TRANSACTION_CHANNEL_NAME,
+      });
+    }
+  } catch (error) {
+    console.error(`Error creating channel for push notification: ${error.message}`, error);
+  }
+};
+
+/**
+ * This function is called when the wallet is initialized with success.
+ */
+export function* onAppInitialization() {
+  yield take(types.START_WALLET_SUCCESS);
+
+  const { enabled, showAmountEnabled } = STORE.getItem(pushNotificationKey.settings) || { enabled: false, showAmountEnabled: false };
+  const hasBeenEnabled = STORE.getItem(pushNotificationKey.hasBeenEnabled);
+  const enabledAt = STORE.getItem(pushNotificationKey.enabledAt);
+
+  const persistedDeviceId = STORE.getItem(pushNotificationKey.deviceId);
+  const deviceId = yield call(getDeviceId);
+  // If the deviceId is different from the persisted one, we should update it.
+  // The first time the perisistedDeviceId will be null, and the deviceId will be
+  // the one returned by getDeviceId, which gets the deviceId from FCM.
+  if (deviceId && persistedDeviceId !== deviceId) {
+    STORE.setItem(pushNotificationKey.deviceId, deviceId);
+    yield put(pushUpdateDeviceId({ deviceId }));
+  }
+
+  // Initialize the pushNotification state on the redux store
+  yield put(pushInit({
+    deviceId,
+    settings: {
+      enabled,
+      showAmountEnabled,
+    },
+    hasBeenEnabled,
+    enabledAt,
+  }));
+
+  yield call(createChannelIfNotExists);
```

Comment:
If any of these requests below fail, the push notification feature won't work, right? Right now you have a `try...catch` but you are not doing anything to handle it. How will the app handle when those requests fail?

Reply:
Hmmm. I will halt the push initialization, and in the feature toggle PR I will complement by disabling the push notification.